CRM/Contact/Import/ImportJob.php

```diff
@@ -68,7 +68,7 @@ public function __construct($tableName = NULL, $createSql = NULL, $createTable =
         $tableName = 'civicrm_import_job_' . md5(uniqid(rand(), TRUE));
       }
       $db->query("DROP TABLE IF EXISTS $tableName");
-      $db->query("CREATE TABLE $tableName ENGINE=InnoDB DEFAULT CHARACTER SET utf8 COLLATE utf8_unicode_ci $createSql");
+      $db->query("CREATE TABLE $tableName ENGINE=InnoDB ROW_FORMAT=Dynamic DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci $createSql");
```

Comment:
I don't think we should be adding to what we hard-code for a table definition - this seems like something that needs to be set at the server level

Reply:
My 2 cents is - CiviCRM should specify whatever it needs to in SQL, so that it can function fine on older (but still supported) MySQL versions.

ROW_FORMAT=Dynamic is the default config on MySQL 5.7, so it will soon not be necessary,  but is already specified in some parts of the code, as it's needed for MySQL 5.6 to work - see CRM_Core_BAO_CustomField::defaultCustomTableSchema()

Note I'm actually in favor of dropping support for MySQL 5.6, but I included this line just in case it will still be supported for some time.

Reply:
@mfb so it's being added FOR 5.6? We shouldn't make any changes to support 5.6 at this stage as it has only a few releases left to go.

Reply:
Yes a fair amount of this PR was to make things work on MySQL 5.5 and 5.6, where the default configs don't work.  On 5.7 it just works so things are pretty simple.

Reply:
@mfb OK - let's pull out the 5.5 & 5.6 stuff - 5.5 is even officially unsupported now - and we are not far off with 5.6. However the definition of supported is also 'we don't break stuff for you' and I think in general we want to target new installs & DBs choosing to convert anyway. If we remove the ROW_FORMAT stuff as well as the on upgrade stuff & the merged stuff this should be starting to look fairly trim